### DISCUSSION TITLE
How do I move all items from left TabBar (Explorer, Search, etc.) to the right TabBar (Outline)

### DISCUSSION THREAD

**Discussion by thers** [eclipse-theia/theia#8600] [Q&A] [answered]
So I need to move all the left TabBar items to the right one in order to get rid of that left bar, how would I achieve that?

**Comment by vince-fugnitto** [comment, marked as answer]
@thers as far as I can tell, you would need to `rebind` these components and open them to the right rather than the left, but it does not prevent future extensions, views, widgets from being added to the left. It does not also prevent end-users from moving views themselves to any location in the application.

You may try to update the `sidepanel` code to never create the `left` panel, and if views request to be opened in the `left` you can instead add them to the `right`.

**Reply by thers** [reply]
Yeah at first I tried rebinding, but it's too much code copy-paste, which isn't good for future updates, so instead I'm money-patching contributions of `FrontendApplicationContribution` that do extend `AbstractViewContribution` and if such has `defaultWidgetOptions.area === 'left'` I change that to right, ofc it's a total "meh" way as I am monkey-patching `private` property, but at least this doesn't lead to massive code copy-paste.

**Reply by thers** [reply]
Alternatively, I did extend `ApplicationShell` with it's further rebinding, overridding `addWidget`, that's a better option than monkey-patching, haha.
